README.md

```diff
@@ -6,55 +7,297 @@ This project aims to build a non-conventional alarm clock with smart features su
 
 [GSoC page](https://summerofcode.withgoogle.com/programs/2023/projects/c7GIl1mf)
 
-## Getting Started
+## Table of Contents
+- [GetX Pattern](#getx-pattern)
+- [Database Schema](#database-schema)
+- [Installation & Setup](#installation-setup)
+- [User Interface & Features](#user-interface-features)
+- [Contribution Guidelines](#contribution-guidelines)
+- [Future Plans](#future-plans)
+- [Community](#community)
+- [Flutter](#flutter)
+
+## GetX Pattern
+
+The "Ultimate Alarm Clock" project employs the GetX pattern for state management. The GetX pattern is a popular state management solution in the Flutter ecosystem, known for its simplicity, efficiency, and developer-friendly approach. It simplifies the process of managing the state of a Flutter application and helps in building reactive and performant user interfaces.
+
+To learn more about the GetX, you can read the documentation [_here_](https://chornthorn.github.io/getx-docs/).
+
+## Database Schema
+
+The "Ultimate Alarm Clock" project utilizes multiple databases for different purposes. These databases include Firebase Firestore, ISAR, and Flutter Secure Storage. Each database serves a distinct role in managing various aspects of the application. Below we provide an overview of the schema for each of these databases. 
+
+### Firebase Firestore
+
```

Comment:
Should probably make it clearer why we have 2 databases. Firestore is necessary for shared alarms, IsarDb is necessary for offline alarms

Reply:
Also, we need to talk about how the shared alarms is setup using Firebase since I think that's the most confusing part

Reply:
Right, I should have touched upon that part. I'll be doing that now.